### PR TITLE
Feat mod star

### PR DESCRIPTION
**Pull request by sidorov-si**
STAR module finished: genome indexing and read mapping

### CODE REVIEW

File: configs/module-defaults.config [hunk 1]
```diff
@@ -127,6 +127,22 @@ params {
...
+            varVCFfile           = ""
+        } 
+        'star_alignReads' {
+            args                 = ""
+            publish_dir          = "star_alignReads"
+            publish_results      = "all"
+            sjdbGTFfile          = ""
+            sjdbFileChrStartEnd  = ""
+            varVCFfile           = ""
+        }
```
Comment (chris-cheshire) [inline configs/module-defaults.config:145]: Can you remove camel case to keep the naming convention? `star_genome_generate`

File: tools/star/main.nf [hunk 2]
```diff
@@ -1,69 +1,153 @@
...
 
 // Specify DSL2
-nextflow.preview.dsl = 2
+nextflow.enable.dsl=2
 
-// Process definition
-process star_map {
-    tag "${sample_id}"
+// Generate genome index
+process star_genomeGenerate {
```
Comment (chris-cheshire) [inline tools/star/main.nf:7]: naming conventions here

File: tools/star/main.nf [hunk 3]
```diff
@@ -1,69 +1,153 @@
...
+
+    // Run read mapping with STAR
+    """
+    mkdir genome_index
+    ${index_command}
+    """
+}
+
+// Map reads
+process star_alignReads {
```
Comment (chris-cheshire) [inline tools/star/main.nf:77]: naming convention

File: configs/module-defaults.config (hunk 1 above)
Comment (sidorov-si) [reply configs/module-defaults.config:145]: Thank you Chris, great points! I'll combine the tests then and remove camel case.

**Comment by sidorov-si** [comment]
Hey @chris-cheshire ! Could you have a look at the reviewed version of the module?
The changes are:
1) combination of all tests in one main.nf;
2) introduction of a custom config file to make possible calling of the same module several times with different publish directories;
3) replacement of our STAR container with a biocontainer with the latest version of STAR;
4) removal of the camel case everywhere;
5) removal of the "report" and "chrNameFile" output channels in the indexing module: output of the chromosome names separately is not so useful as I thought at the beginning, and the "report" channel from the indexing module doesn't make much sense (I don't think MultiQC would be able to parse it). Also, I guessed from a poorly reproducible error that I had before that publishing the whole index dir and then a log file from this dir into the same publish dir sometimes caused a race condition, and the index dir was not publised entirely.

**Pull request merged**